src/common/settings.js

```diff
@@ -58,32 +58,31 @@ const merges =
 
 export const fetch =
   (names/*:Array<Name>*/)/*:Task<Never, Result<Error, Settings>>*/ =>
-  Task.future(() => {
+  Task.create(resolve => {
     if (navigator.mozSettings != null) {
       const lock = navigator.mozSettings.createLock();
       const settings = names.map(name => lock.get(name));
-
-      return Promise.all(settings)
-                    .then(merges)
-                    .then(ok, error);
+      Promise.all(settings)
```

Comment:
Nit: Could you please move `.all(settings)` on the next line as well since you're changing indentation anyhow.